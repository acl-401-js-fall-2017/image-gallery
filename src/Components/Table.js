import React, { PureComponent } from 'react';
import { PropTypes } from 'prop-types';
import { removeImage } from '../Actions/image.actions';

export default class Table extends PureComponent {
  render() {
    const { images } = this.props;
    return (
      <table>
        <tbody>
          <tr>
            <th>Title</th>
            <th>Description</th>
          </tr>
          {images.map((image, i) => {
           
            return (
              <tr className="table-row" key={i}>
                <td><a href={image.url}>{image.title}</a></td>
                <td>{image.description}</td>
                <td>
                  <button onClick={() => removeImage(image._id)}> Remove </button>
                </td>
                
              </tr>
            );
          })}
        </tbody>
      </table>
    );
  }
}

Table.propTypes = {
  images: PropTypes.array
};