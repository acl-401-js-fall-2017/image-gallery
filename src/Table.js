import React, { PureComponent } from 'react';

class Table extends PureComponent {
  render() {
    const { images } = this.props;
    return (
      <table>
        <tr>
          <th>Title</th>
          <th>Description</th>
        </tr>
        {images.map((image, i) => {
          return (
            <tr className="table-row" key={i}>
              <td><a href={image.url}>{image.title}</a></td>
              <td>{image.description}</td>
            </tr>
          );
        })}
      </table>
    );
  }
}



export default Table;