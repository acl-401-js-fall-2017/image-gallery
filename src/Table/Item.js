import React, { Component } from 'react';

export default class Item extends Component {
  render() {
    const { image } = this.props;
    return (
      <tr className="Item">
        <td>{image.title}</td>
        <td>{image.description}</td>
        <td>{image.url}</td>
      </tr>
    );
  }
}