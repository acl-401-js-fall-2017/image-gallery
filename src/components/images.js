import React, { Component } from 'react';
import '../App.css';
import shortid from 'shortid';
import WonderList from '../views/list';
import WonderThumbnail from '../views/thumbnail';
import WonderGallery from '../views/gallery';
import AddImage from '../AddImage';
import { createWonder } from '../actions';
import {
  Route,
  Switch
} from 'react-router-dom';


class Images extends Component {
  constructor(){
    super();
    this.state = {
      view: 'list',
      wonders: [
        {
          _id: shortid.generate(),
          title: 'Great Pyramid of Giza',
          description : 'The Great Pyramid of Giza is the oldest and largest of the three pyramids in the Giza pyramid complex bordering what is now El Giza, Egypt. It is the oldest of the Seven Wonders of the Ancient World, and the only one to remain largely intact.',
          url:'https://upload.wikimedia.org/wikipedia/commons/e/e3/Kheops-Pyramid.jpg'
        },
        {
          _id: shortid.generate(),
          title: 'Great Wall of China',
          description : 'The Great Wall of China is a series of fortifications made of stone, brick, tamped earth, wood, and other materials, generally built along an east-to-west line across the historical northern borders of China to protect the Chinese states and empires against the raids and invasions of the various nomadic groups of the Eurasian Steppe.',
          url:'https://upload.wikimedia.org/wikipedia/commons/6/6f/The_Great_Wall_of_China_at_Jinshanling.jpg'
        },
        {
          _id: shortid.generate(),
          title: 'Petra',
          description : 'Petra, originally known to the Nabataeans as Raqmu, is a historical and archaeological city in southern Jordan. The city is famous for its rock-cut architecture and water conduit system. Another name for Petra is the Rose City due to the color of the stone out of which it is carved.',
          url:'https://upload.wikimedia.org/wikipedia/commons/b/b7/The_Monastery%2C_Petra%2C_Jordan8.jpg'
        },
        {
          _id: shortid.generate(),
          title: 'The Colosseum',
          description : 'The Colosseum or Coliseum , also known as the Flavian Amphitheatre (Latin: Amphitheatrum Flavium; Italian: Anfiteatro Flavio [aŋfiteˈaːtro ˈflaːvjo] or Colosseo [kolosˈsɛːo]), is an oval amphitheatre in the centre of the city of Rome, Italy. Built of concrete and stone,[1] it is the largest amphitheatre ever built.',
          url:'https://upload.wikimedia.org/wikipedia/commons/f/f7/KOLOSSEUM_-_panoramio.jpg'
        },
        {
          _id: shortid.generate(),
          title: 'Chichen Itza',
          description : 'Chichen Itza was one of the largest Maya cities, with the relatively densely clustered architecture of the site core covering an area of at least 5 square kilometres (1.9 sq mi). Smaller scale residential architecture extends for an unknown distance beyond this. The city was built upon broken terrain, which was artificially levelled in order to build the major architectural groups, with the greatest effort being expended in the levelling of the areas for the Castillo pyramid, and the Las Monjas, Osario and Main Southwest groups.',
          url:'https://upload.wikimedia.org/wikipedia/commons/7/7a/Chichen-Itza-Castillo-Seen-From-East.JPG'
        },
        {
          _id: shortid.generate(),
          title: 'Machu Picchu',
          description : 'Most archaeologists believe that Machu Picchu was constructed as an estate for the Inca emperor Pachacuti (1438–1472). Often mistakenly referred to as the "Lost City of the Incas" (a title more accurately applied to Vilcabamba), it is the most familiar icon of Inca civilization. The Incas built the estate around 1450 but abandoned it a century later at the time of the Spanish Conquest',
          url:'https://upload.wikimedia.org/wikipedia/commons/e/eb/Machu_Picchu%2C_Peru.jpg'
        },
        {
          _id: shortid.generate(),
          title: 'Taj Mahal',
          description : 'The Taj Mahal is an ivory-white marble mausoleum on the south bank of the Yamuna river in the Indian city of Agra. It was commissioned in 1632 by the Mughal emperor, Shah Jahan (reigned 1628–1658), to house the tomb of his favourite wife, Mumtaz Mahal. The tomb is the centrepiece of a 17-hectare (42-acre) complex, which includes a mosque and a guest house, and is set in formal gardens bounded on three sides by a crenellated wall.',
          url:'https://upload.wikimedia.org/wikipedia/commons/c/c8/Taj_Mahal_in_March_2004.jpg'
        }
      ]
    };
    this.handleOnSubmitCreateWonder = this.handleOnSubmitCreateWonder.bind(this);
  }

 
  handleOnSubmitCreateWonder = (title, description, url) => {
    const newState = createWonder(this.state, title, description, url);
    this.setState(newState);
  }
  

  render() {
    const { wonders } = this.state;
    return (
      <div className="App">
        <header className="App-header">
          <h1 className="App-title">Wonders of The World!!!</h1>
        </header>
        <div>
          <Switch>
            <Route exact path={`${this.props.match.url}/list`} render={() => <WonderList wonders={wonders}/>}/>
            <Route exact path={`${this.props.match.url}/list`} render={() => <WonderThumbnail wonders={wonders}/>}/>
            <Route exact path={`${this.props.match.url}/:gallery?`}  render={() => <WonderGallery wonders={wonders}/>}/>
          </Switch>
        </div>
        <AddImage handleOnSubmit={this.handleOnSubmitCreateWonder}/>
      </div>
    );
  }
}

export default Images;
