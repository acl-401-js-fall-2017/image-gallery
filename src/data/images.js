const imageData = [
  {
    title: 'Culprit Apprehended',
    description: 'Milk was left home alone.',
    url: '/images/20160310_214017.jpg'
  },
  {
    title: 'Sunset on Marine Drive',
    description: 'Milk joined us for some longboarding by Marine Dr.',
    url: '/images/20160416_194337.jpg'
  },
  {
    title: 'First Meal Together',
    description: 'Oreo and Milk sharing their first meal together after a tumultuous first meeting.',
    url: '/images/20160527_132115.jpg'
  }
];

export default imageData;