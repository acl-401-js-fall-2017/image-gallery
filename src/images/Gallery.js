import React, { PureComponent } from 'react';
import styled from 'styled-components';
import { loadGallery, deleteImg } from '../utils/actions';
import imgApi from '../utils/services/imgApi';


export default class Gallery extends PureComponent {
  state = { 
    index: 0, 
    gallery:[]
  }

  async componentDidMount() {
    const img = await imgApi.get();
    const newState = loadGallery(this.state, img);
    this.setState(newState);
  }

  handleDelete = async id => {
    await imgApi.remove(id);
    const newState = deleteImg(this.state, id);
    this.setState(newState);
  }


  handleClick = (value) => {
    console.log('clicked');
    const newState = {
      ...this.state,
      index: this.state.index + value
    };
    this.setState(newState);
  }

  render() {
    return( 
      <StyledDiv>
        <div>
          {this.state.gallery.map((img, i) => (
            <ImgDiv key={img._id} shouldDisplay ={this.state.index === i}>
              <span> {img.title} </span>
              <DeleteDiv onClick={() => this.handleDelete(img._id)}>X</DeleteDiv>
              <img style={{ width:'100%' }} src={img.url} alt=''/>
              <span> {img.description} </span>
              { i !== this.state.gallery.length -1 && <span onClick ={()=> this.handleClick(1)}> next </span>}
              {i !== 0 && <span onClick ={()=> this.handleClick(-1)}> previous</span>}
            </ImgDiv>
          ))}
        </div>
      </StyledDiv>
    );
  }
}

const StyledDiv = styled.div`
display: 'flex';
`;

const ImgDiv = styled.div`
display:${props => props.shouldDisplay ? 'flex' : 'none'};
flex-direction: column;
margin: 0 10%;
`;

const DeleteDiv = styled.div`
margin-left: 90%;
text-align: center;
border: 1px solid black;
`;