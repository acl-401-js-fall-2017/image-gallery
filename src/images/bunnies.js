//import shortid from 'shortid';

const Bunnies = [
  {
    _id: '1',
    title: 'Best Bunny',
    description: 'Just the cutest color on a bunny ever',
    url: 'https://s3.amazonaws.com/f.cl.ly/items/3g3J1G0w122M360w380O/3726490195_f7cc75d377_o.jpg'
  },
  {
    _id: '2',
    title: 'Kind of cute Bunny',
    description: 'Too many carbs :(',
    url: 'http://static.boredpanda.com/blog/wp-content/uploads/2015/09/cute-bunnies-25__605.jpg'
  },
  {
    _id: '3',
    title: 'Tiny Bunny',
    description: 'Cant see me',
    url: 'http://static.boredpanda.com/blog/wp-content/uploads/2015/09/cute-bunnies-110__605.jpg'
  }
];

export default Bunnies;