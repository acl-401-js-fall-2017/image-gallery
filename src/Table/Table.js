import React, { Component } from 'react';
import Item from './Item';

class Table extends Component {
  render() {
    const { images } = this.props;
    return (
      <table>
        <tr>
          <th>Title</th>
          <th>Link</th>
          <th>Description</th>
        </tr>
        {images.map(image => <Item image={image}/>)}
      </table>
    );
  }
}