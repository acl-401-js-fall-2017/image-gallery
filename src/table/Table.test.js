import React from 'react';

import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import images from '../images';
import Table from './Table';

describe('Table', () => {
  it('table of images', () => {
    const wrapper = shallow(<Table images={images}/>);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});