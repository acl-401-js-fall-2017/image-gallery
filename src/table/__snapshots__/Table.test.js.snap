// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table table of images 1`] = `
<div>
  <table>
    <tbody>
      <tr>
        <th>
          Title
        </th>
        <th>
          Description
        </th>
      </tr>
      <tr
        className="table-row"
        key="0"
      >
        <td>
          <a
            href="http://f.cl.ly/items/3g3J1G0w122M360w380O/3726490195_f7cc75d377_o.jpg"
          >
            Cute Bunny
          </a>
        </td>
        <td>
          Isn't it fuzzy-wuzzy cutest thing you've ever seen?
        </td>
        <td>
          <button
            onClick={[Function]}
          >
             Remove 
          </button>
        </td>
      </tr>
      <tr
        className="table-row"
        key="1"
      >
        <td>
          <a
            href="http://static.boredpanda.com/blog/wp-content/uploads/2015/09/cute-bunnies-25__605.jpg"
          >
            Cute Bunn
          </a>
        </td>
        <td>
          Isn't it fuzzy-wuzzy cutest thing you've ever seen?
        </td>
        <td>
          <button
            onClick={[Function]}
          >
             Remove 
          </button>
        </td>
      </tr>
      <tr
        className="table-row"
        key="2"
      >
        <td>
          <a
            href="http://static.boredpanda.com/blog/wp-content/uploads/2015/09/cute-bunnies-110__605.jpg"
          >
            Cute Bun
          </a>
        </td>
        <td>
          Isn't it fuzzy-wuzzy cutest thing you've ever seen?
        </td>
        <td>
          <button
            onClick={[Function]}
          >
             Remove 
          </button>
        </td>
      </tr>
    </tbody>
  </table>
  <AddImages />
</div>
`;
